### DISCUSSION TITLE
Can't use absolute paths in api folder

### DISCUSSION THREAD

**Discussion by user72356** [vercel/next.js#42447] [Help]
Test repo at https://github.com/user72356/vite-project

I'm trying to import modules using their absolute path from within the api directory of my app. I'm able to use absolute paths in the front-end portion of the code, for example `import hello from "shared/hello";`, but not within the api directory.

Doing so results in a `Cannot find module` error. See https://github.com/user72356/vite-project/blob/main/api/index.ts to see the actual code.

I have limited experience with Vercel and Node.js but I think I may understand what's going on. Since anything in the api directory runs on Node, I suppose `shared/hello` doesn't map to anything on the filesystem. It only properly maps to something on the web, i.e. example.com/shared/hello. Let me know if I got this right.

How can I make `shared/hello` be a valid import path on the front-end as well as on the back-end (api folder). Is this even possible?

**Comment by icyJoseph** [comment]
Hi,

That last statement is misleading. When your code is compiled, the paths are resolved proper. I have API routes which use absolute paths no problem, in dev and prod modes.

Could you share a repository? Otherwise we'll go down a series of shots in the dark trying to figure out why this is not working for you.

**Reply by user72356** [reply]
Hi, thank you for replying. I actually rewrote my message based on your feedback and added a repo. I also gained a little more insight into the import process with the api folder (I think?), but still looking for a solution as outlined in the OP. Thank you!

**Reply by icyJoseph** [reply]
Aha, this was a Vite project. I think the issue might be with `@vercel/node` which doesn't directly support path mappings:

- https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/node-js#using-typescript-with-the-node.js-runtime
- https://www.typescriptlang.org/docs/handbook/module-resolution.html#path-mapping

> You can also use a tsconfig.json file at the root of your project to configure the TypeScript compiler. Most options are supported aside from ["Path Mappings"](https://www.typescriptlang.org/docs/handbook/module-resolution.html#path-mapping) and ["Project References"](https://www.typescriptlang.org/docs/handbook/project-references.html).

I guess your API route will have to use relative imports for the time being. I used `vercel dev --debug` to quickly point to `@vercel/node` and then sat down to read the docs in the Vercel website. Pretty sure this is the issue.

**Reply by user72356** [reply]
I'm actually using `base-url` which is not explicitely unsupported: https://www.typescriptlang.org/docs/handbook/module-resolution.html#base-url

> Setting [baseUrl](https://www.typescriptlang.org/tsconfig#baseUrl) informs the compiler where to find modules. All module imports with non-relative names are assumed to be relative to the [baseUrl](https://www.typescriptlang.org/tsconfig#baseUrl).

So I'm not sure whether it falls under what you describe?

**Reply by icyJoseph** [reply]
Well, if they did support it, it'd work 😟
